src/commands/init.js

```diff
@@ -83,84 +222,323 @@ Examples:
       return
     }
 
-    let installCommand = yarn ? 'yarn' : 'npm install'
-    let codegenCommand = yarn ? 'yarn codegen' : 'npm run codegen'
-    let deployCommand = yarn ? 'yarn deploy' : 'npm run deploy'
+    let commands = {
+      install: yarn ? 'yarn' : 'npm install',
+      codegen: yarn ? 'yarn codegen' : 'npm run codegen',
+      deploy: yarn ? 'yarn deploy' : 'npm run deploy',
+    }
 
-    // Clone the example subgraph repository
-    let spinner
-    try {
-      spinner = print.spin('Cloning example subgraph')
-      await system.run(
-        `git clone http://github.com/graphprotocol/example-subgraph ${directory}`
+    // If all parameters are provided from the command-line,
+    // go straight to creating the subgraph from the example
+    if (fromExample && subgraphName && directory) {
+      return await initSubgraphFromExample(
+        toolbox,
+        { allowSimpleName, directory, subgraphName },
+        { commands }
       )
-      spinner.stop()
-    } catch (e) {
-      spinner.fail(`Failed to clone example subgraph repository: ${e}`)
-      process.exitCode = 1
-      return
     }
 
-    // Update package.json to match the subgraph name
-    try {
-      // Load package.json
-      let pkgJsonFilename = filesystem.path(directory, 'package.json')
-      let pkgJson = await filesystem.read(pkgJsonFilename, 'json')
+    // If all parameters are provided from the command-line,
+    // go straight to creating the subgraph from an existing contract
+    if (fromContract && subgraphName && directory && network) {
+      let abi
+      try {
+        abi = await loadAbiFromEtherscan(network, fromContract)
+      } catch (e) {
+        process.exitCode = 1
+        return
+      }
 
-      pkgJson.name = getSubgraphBasename(subgraphName)
-      Object.keys(pkgJson.scripts).forEach(name => {
-        pkgJson.scripts[name] = pkgJson.scripts[name].replace('example', subgraphName)
-      })
-      delete pkgJson['license']
-      delete pkgJson['repository']
+      return await initSubgraphFromContract(
+        toolbox,
+        { abi, allowSimpleName, directory, address: fromContract, network, subgraphName },
+        { commands }
+      )
+    }
 
-      // Write package.json
-      await filesystem.write(pkgJsonFilename, pkgJson, { jsonIndent: 2 })
-    } catch (e) {
-      print.error(`Failed to preconfigure the subgraph: ${e}`)
-      process.exitCode = 1
-      filesystem.remove(directory)
+    // Otherwise, take the user through the interactive form
+    let inputs = await processInitForm(toolbox, {
+      abi,
+      allowSimpleName,
+      directory,
+      address: fromContract,
+      fromExample,
+      network,
+      subgraphName,
+    })
+
+    if (inputs === undefined) {
+      process.exit(1)
```

Comment:
Does `inputs === undefined` have a particular meaning? Should we print something here?

Reply:
The `inputs` are `undefined` if the form is cancelled. I've added a comment.

Reply:
Where's the comment?

Reply:
https://github.com/graphprotocol/graph-cli/blob/jannis/graph-init-with-two-modes/src/commands/init.js#L281 😉